### PR TITLE
Add documentation on Opcode Assertions method

### PR DESCRIPTION
**Pull request by toni-calvin**
Add documentation on Opcode Assertions method

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -606,6 +606,83 @@ This method will continuously execute cairo steps until the end pc, returned by
...
+
+*Compute operands*
+
+```go
+    //TODO
+```
+
+*Opcode assertions*
+
+Once we have the instruction's operands to work with, we have to ensure the correctness of them. The first thing we need to differenciate is which type of instruction are we running, we do this by switching the `opcode`.
```
Comment (fmoletta) [inline README.md]: > we do this by switching the `opcode`. 

This line is confusing, we could instead say, by looking at the instruction's opcode

File: README.md [hunk 2]
```diff
@@ -606,6 +606,83 @@ This method will continuously execute cairo steps until the end pc, returned by
...
+
+```go
+    //TODO
+```
+
+*Opcode assertions*
+
+Once we have the instruction's operands to work with, we have to ensure the correctness of them. The first thing we need to differenciate is which type of instruction are we running, we do this by switching the `opcode`. 
+
+The posible options (for now) are:
```
Comment (fmoletta) [inline README.md]: This is not correct, there are no other options waiting to be added, we could instead say The options we care about in this method/ The options we want to perform assertions on

Comment (fmoletta) [reply README.md]: Also we could refer to the as opcodes

File: README.md [hunk 3]
```diff
@@ -606,6 +606,83 @@ This method will continuously execute cairo steps until the end pc, returned by
...
+
+Once we have the instruction's operands to work with, we have to ensure the correctness of them. The first thing we need to differenciate is which type of instruction are we running, we do this by switching the `opcode`. 
+
+The posible options (for now) are: 
+	1. AssertEq instruction 
+	2. Call instruction 
+
+In the first option, we need to ensure the result operand is not null (nil in this case) and also that the result operand is equal to the destiny operand. If any of those things fail, we throw an error. 
+
+On the other hand, the Call instruction, what we do first is define our return PC register, we do that adding the size of the instruction to the current PC. Then, we check our operand OP0 is equal to the return PC and our destiny operand is the same as the return FP register. If any of those things fail, we throw an error.
```
Comment (fmoletta) [inline README.md]:
```suggestion
On the other hand, the Call instruction, what we do first is define our return PC register, we do that adding the size of the instruction to the current PC. Then, we check our operand OP0 is equal to the return PC and our destiny operand is the same as the return FP register. If any of those things fail, we throw an error. 
```
destiny operand doesn't sound quite right, we should use destination, or just dst instead

File: README.md [hunk 4]
```diff
@@ -606,6 +606,83 @@ This method will continuously execute cairo steps until the end pc, returned by
...
+
+The posible options (for now) are: 
+	1. AssertEq instruction 
+	2. Call instruction 
+
+In the first option, we need to ensure the result operand is not null (nil in this case) and also that the result operand is equal to the destiny operand. If any of those things fail, we throw an error. 
+
+On the other hand, the Call instruction, what we do first is define our return PC register, we do that adding the size of the instruction to the current PC. Then, we check our operand OP0 is equal to the return PC and our destiny operand is the same as the return FP register. If any of those things fail, we throw an error. 
+
+If this method returns nil, it means operands are correct and we are good to go!
```
Comment (fmoletta) [inline README.md]:
```suggestion
If this method returns a nil error, it means operands were computed correctly and we are good to go!
```

File: README.md (hunk 1 above)
Comment (toni-calvin) [inline README.md]:
```suggestion
Once we have the instruction's operands to work with, we have to ensure the correctness of them. The first thing we need to differentiate is which type of instruction are we running, we do this by looking at the instruction's opcode. 
```

Comment (toni-calvin) [reply README.md]: Ty!

File: README.md (hunk 2 above)
Comment (toni-calvin) [inline README.md]:
```suggestion
The posible opcodes we want to perform assertions on are: 
```

Comment (toni-calvin) [reply README.md]: ty!

File: README.md (hunk 3 above)
Comment (toni-calvin) [reply README.md]: I agree!

Comment (toni-calvin) [inline README.md]:
```suggestion
On the other hand, the Call instruction, what we do first is define our return pc register, we do that adding the size of the instruction to the current pc. Then, we check our operand op0 is equal to the return pc and our dst operand is the same as the return fp register. If any of those things fail, we throw an error. 
```

File: README.md [hunk 5]
```diff
@@ -606,6 +606,83 @@ This method will continuously execute cairo steps until the end pc, returned by
...
+
+*Opcode assertions*
+
+Once we have the instruction's operands to work with, we have to ensure the correctness of them. The first thing we need to differentiate is which type of instruction are we running, we do this by looking at the instruction's opcode. 
+
+The posible opcodes we want to perform assertions on are: 
+	1. AssertEq instruction 
+	2. Call instruction 
+
+In the first option, we need to ensure the result operand is not null (nil in this case) and also that the result operand is equal to the destiny operand. If any of those things fail, we throw an error.
```
Comment (toni-calvin) [inline README.md]:
```suggestion
In the first option, we need to ensure the result operand is not null (nil in this case) and also that the result operand is equal to the dst operand. If any of those things fail, we throw an error. 
```

**Review by fmoletta — APPROVED** [review]

**Review by pablodeymo — APPROVED** [review]
👍 🚀

**Pull request merged**